drivers/volume/portworx/portworx.go

```diff
@@ -1517,6 +1550,40 @@ func (d *portworx) getKvdbMembers(n node.Node) (map[string]metadataNode, error)
 	return kvdbMembers, err
 }
 
+func (d *portworx) CollectDiags(n node.Node) error {
+	var err error
+
+	pxNode, err := d.getPxNode(n, nil)
+	if err != nil {
+		return err
+	}
+	if pxNode.Status == api.Status_STATUS_OFFLINE {
```

Comment:
@vilasd when PX is offline, shouldn't we still take diags? We won't be able to take live cores but we can atleast take the offline diags.

Reply:
Yes! put else condition to collect regular diags

Reply:
As per conversation with @jrivera-px, if the container is not running 9014 won't be there so you won't be able to issue the request. They have same check in Porx for collection diags.

Reply:
The diags server runs inside the PX container.   So if the PX container is down/not running you won't be able to issue a diags via an http request.   In that case you should take the diags from the host which px is running on.

Reply:
Ok I will use pxctl sv diags to collect it incase of PX is offline.